packages/vscode-extension/src/commands/ListFiles.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ ** Copyright (c) 2021 Oracle and/or its affiliates.  All rights reserved.
+ ** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+ */
+import { window } from 'vscode';
+import { COMMAND, LIST_FILES } from '../service/TranslationKeys';
+import { actionResultStatus, FolderUtils } from '../util/ExtensionUtil';
+
+import BaseAction from './BaseAction';
+import { FolderType } from './FolderType';
+
+const COMMAND_NAME = 'listfiles';
+
+const LIST_FILES_COMMAND = {
+	OPTIONS: {
+		FOLDER: 'folder',
+	},
+};
+
+export default class ListFiles extends BaseAction {
+	constructor() {
+		super(COMMAND_NAME);
+	}
+
+	protected async execute(): Promise<void> {
+		const fileCabinetFolders = await this._getListFolders();
+		const filteredFolders = this.filterDisabledFolders(fileCabinetFolders);
+		if (filteredFolders === undefined) {
+			return;
+		}
+
+		const selectedFolder = await this._selectFolder(filteredFolders);
+
+		if (selectedFolder === undefined) {
+			return;
+		}
+
+		await this._listFiles(selectedFolder);
+	}
+
+	protected async _getListFolders() {
+		const listFoldersPromise = FolderUtils.getFileCabinetFolders(this.executionPath, COMMAND_NAME);
+		const statusBarMessage = this.translationService.getMessage(LIST_FILES.LOADING_FOLDERS);
+		this.messageService.showStatusBarMessage(statusBarMessage, listFoldersPromise);
+
+		return listFoldersPromise;
+	}
+
+	private filterDisabledFolders(fileCabinetFolders: any) {
+		return fileCabinetFolders.filter((folder: FolderType) => folder.disabled === '');
+	}
+
+	// protected async _selectFolder(filteredFolders: { map: (key: (folder: FolderType) => string | undefined) => string[] }): Promise<string | undefined> {
+	protected async _selectFolder(filteredFolders: {
+		map: (arg0: (folder: FolderType) => string | undefined) => string[];
+	}): Promise<string | undefined> {
```

Comment:
Why the need for undefined here?

Reply:
Typescript is making me adding it. I suppose you could decide to not select any option.  (like pressing Escape). That´s why it is needed to define the possible output of the promise as string or undefined